loguru_mypy/__init__.py

```diff
@@ -152,15 +197,65 @@ def _loguru_logger_call_handler(
     return ctx.default_return_type
 
 
+def _analyze_record_results(
+    log_msg_expr: StrExpr,
+    log_msg_record_references: t.Sequence[str],
+    call_kwargs: t.Mapping[str, Expression],
+    logger_opts: Opts,
+    *,
+    ctx: MethodContext,
+) -> bool:
+    if logger_opts.record and 'record' in call_kwargs:
+        ctx.api.msg.fail(
+            'record keyword argument cannot override record structure',
+            context=log_msg_expr,
+            code=ERROR_BAD_KWARG,
+        )
+        return False
+    if logger_opts.record and not log_msg_record_references:
+        ctx.api.msg.note(
+            'Logger configured with record=True is not using record structure',
+            context=log_msg_expr,
+            code=ERROR_BAD_RECORD,
+        )
+        return False
+    elif not logger_opts.record and log_msg_record_references:
```

Comment:
@Delgan this I believe is also slightly incorrect. It can happen that logger is actually configured without `record=True` but someone decided to use a kwarg with said name, right ? This will not trigger any runtime exception as far as I remember.

Reply:
Yes, indeed! I tested with `logger.info("=> {record}", record="foo")` which does not generate warning (as expected). However, `logger.info("=> {record['foo']}", record={"foo": "bar"})` raises an error on `mypy` while it shouldn't.